Paws4Me/Paws4Me/Data/LocalPetNameViewController.swift

```diff
@@ -56,65 +72,66 @@ class LocalPetViewController: UIViewController {
-      guard let appDelegate = UIApplication.shared.delegate as? AppDelegate else {
-        return
-      }
-
-      let managedContext = appDelegate.persistentContainer.viewContext
-      let entity = NSEntityDescription.entity(forEntityName: "Pet", in: managedContext)!
-      let pet = NSManagedObject(entity: entity, insertInto: managedContext)
-        pet.setValue(name, forKeyPath: "petName")
-        pet.setValue(imagePet, forKeyPath: "petImage")
-
-      do {
-        try managedContext.save()
-        pets.append(pet)
-      } catch _ as NSError {
-          displayAlert(alertTitle: "Unable to save \(namePet)",
-                             alertMessage: "There was a problem saving",
-                             alertActionTitle: "Try again" ,
-                       alertDelegate: self, alertTriggered: .fatalLocalDatabaseAlert)
-      }
-    }
-  }
-
-  // MARK: - UITableViewDataSource
-  extension LocalPetViewController: UITableViewDataSource {
-
-    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
-      return pets.count
-    }
-
-    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
-      let pet = pets[indexPath.row]
-      let cell = tableView.dequeueReusableCell(withIdentifier: "Cell", for: indexPath)
-      cell.textLabel?.text = pet.value(forKeyPath: "petName") as? String
-        guard let imagePetLocal = (pet.value(forKeyPath: "petImage") as? String) else { return UITableViewCell() }
-        cell.imageView?.loadImageFromURL(imageURL: imagePetLocal)
-      return cell
-    }
-      func tableView(_ tableView: UITableView,
-                     commit editingStyle: UITableViewCell.EditingStyle, forRowAt indexPath: IndexPath) {
-
-          guard let appDelegate = UIApplication.shared.delegate as? AppDelegate else {
-            return
-          }
-
-          let managedContext = appDelegate.persistentContainer.viewContext
-
-        if editingStyle == .delete {
-            managedContext.delete(pets[indexPath.row])
-          do {
-            try managedContext.save()
-            tableView.reloadData()
-          } catch _ as NSError {
-              displayAlert(alertTitle: "Unable to delete)",
-                                 alertMessage: "There was a problem deleting this pet",
-                                 alertActionTitle: "Try again" ,
-                           alertDelegate: self, alertTriggered: .fatalLocalDatabaseAlert)
-          }
-        }
-      }
-  }
+}
+// MARK: - UITableViewDataSource
+   extension LocalPetViewController: UITableViewDataSource {
+
+     func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+         guard let countPet = pets?.count else {return 0}
+         return countPet
+     }
+
+     func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+       let pet = pets?[indexPath.row]
```

Comment:
Think you might be looking for `pets?[indexPath.item]` ?

Reply:
The pets?.[indexPath.row] is intentional